server/src/main/java/org/opensearch/transport/ClusterConnectionManager.java

```diff
@@ -226,6 +244,25 @@ public void disconnectFromNode(DiscoveryNode node) {
             // if we found it and removed it we close
             nodeChannels.close();
         }
+        pendingDisconnections.remove(node);
```

Comment:
Should we also remove as part of connection close listener in line 173?

Reply:
Yeah that makes sense since not all disconnect flows will go through this function, good catch.
Ideally this `pendingDisconnects` list would only matter here (since this function is called in applyCommit of node-left flow), but I agree it is much safer to also clear it as part of close listener as well to ensure we don't have any stale entries, similar to how `connectedNodes` is also cleared in both places